src/devices/src/virtio/block/device.rs

```diff
@@ -117,12 +149,7 @@ impl Block {
         is_disk_root: bool,
         rate_limiter: RateLimiter,
     ) -> io::Result<Block> {
-        let mut disk_image = OpenOptions::new()
-            .read(true)
-            .write(!is_disk_read_only)
-            .open(PathBuf::from(&disk_image_path))?;
-
-        let disk_size = disk_image.seek(SeekFrom::End(0))? as u64;
+        let disk_image = DiskProperties::new(disk_image_path, is_disk_read_only)?;
```

Comment:
Would make sense to rename this to disk_properties.

Reply:
Oki, done.